opentreemap/treemap/udf.py

```diff
@@ -358,17 +369,24 @@ def _validate_and_update_choice(
                 {'datatype': [_("Can't change choices "
                                 "on a non-choice field")]})
 
-        if old_choice_value not in datatype['choices']:
+        choices = datatype['choices']
+        protected_choices = datatype.get('protected_choices', [])
+        all_choices = choices + protected_choices
```

Comment:
The problem (not that it's much of a problem at all) is that UDFs and collection UDFs exercise the same code paths for validate, clean, etc. On the plus side, this means that protected choices should just work for collection UDFs, although I have not thoroughly tested it on collections.